### PR TITLE
Internal: Add Google Maps Atomic Widget [EDS-293]

### PR DESCRIPTION
**Pull request by blackcolt**

### CODE REVIEW

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use Elementor\Modules\AtomicWidgets\Base\Atomic_Widget_Base;
+use Elementor\Modules\AtomicWidgets\Controls\Types\Textarea_Control;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Atomic_Google_Maps extends Atomic_Widget_Base {
+    public function get_atomic_controls(): array
+    {
```
Comment (ronenelementor) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:14]:
```suggestion
    public function get_atomic_controls(): array {
```

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                'address'
+            )->set_label(__('Address', 'elementor')),
+            Textarea_Control::bind_to(
+                'zoom'
+            )->set_label(__('Zoom', 'elementor')),
+        ];
+    }
+
+    public function get_name()
+    {
```
Comment (ronenelementor) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:29]:
```suggestion
    public function get_name() {
```

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return 'a-google-maps';
+    }
+    protected function render()
+    {
```
Comment (ronenelementor) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:33]:
```suggestion
    }

    protected function render() {
```

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    <?php
+    }
+
+    public function get_title() {
+        return esc_html__( 'Atomic Google Maps', 'elementor' );
+    }
+
+
+    public function get_icon()
+    {
```
Comment (ronenelementor) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:59]:
```suggestion
    public function get_icon() {
```

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+	public function get_name() {
+		return 'a-google-maps';
+	}
+	protected function render() {
+		$url = 'https://maps.google.com/maps/?q=%1$s&amp;t=m&amp;z=%2$d&amp;output=embed&amp;iwloc=near';
+		$params = [
+			$this->get_settings( 'address' ) ?? esc_html__( 'London Eye, London, United Kingdom', 'elementor' ),
+			$this->get_settings( 'zoom' ) ?? 10,
+			$this->get_settings( 'api_key' ) ?? false,
```
Comment (Omerisra6) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:35]: maybe extract to variables ( and set defaults ) there

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		];
+
+		if ( $params[2] ) {
+			$url = 'https://www.google.com/maps/embed/v1/place?key=%3$s&q=%1$s&amp;zoom=%2$d';
+		}
```
Comment (Omerisra6) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:40]: shouldn't it be the default?

**Review by Omerisra6 — DISMISSED** [review]

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+		?>
+		<iframe loading="lazy"
+				src="<?php echo esc_url( vsprintf( $url, $params ) ); ?>"
+				title="<?php echo esc_attr( $params[0] ); ?>"
+				aria-label="<?php echo esc_attr( $params[0] ); ?>"
```
Comment (Nevoss) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:45]: Fix indent

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+		return [
+			Textarea_Control::bind_to(
+				'api_key'
+			)->set_label( __( 'API Key', 'elementor' ) ),
+			Textarea_Control::bind_to(
+				'address'
+			)->set_label( __( 'Address', 'elementor' ) ),
+			Textarea_Control::bind_to(
+				'zoom'
+			)->set_label( __( 'Zoom', 'elementor' ) ),
```
[the remark this answers is not in the document]
Comment (blackcolt) [reply modules/atomic-widgets/widgets/atomic-google-maps.php:23]: wil do, thanks.

File: modules/atomic-widgets/widgets/atomic-google-maps.php (hunk 5 above)
Comment (blackcolt) [reply modules/atomic-widgets/widgets/atomic-google-maps.php:35]: why??!

File: modules/atomic-widgets/widgets/atomic-google-maps.php (hunk 6 above)
Comment (Nevoss) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:40]: Not clear, try to make the code more readable, use variable for if staments, do the spring right away.

something like that:
```php 
$url = $has_api_key ? '...' : '...';

$url = sprintf($url, [....HERE_ALL_PARAMS]);
```

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+			$this->get_settings( 'zoom' ) ?? 10,
+			$this->get_settings( 'api_key' ) ?? false,
+		];
+
+		if ( $params[2] ) {
+			$url = 'https://www.google.com/maps/embed/v1/place?key=%3$s&q=%1$s&amp;zoom=%2$d';
+		}
+		?>
+		<iframe loading="lazy"
+				src="<?php echo esc_url( vsprintf( $url, $params ) ); ?>"
```
Comment (Nevoss) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:43]: Whats the diff between `vsprintf` to `sprintf`

**Review by Nevoss — DISMISSED** [review]
We should wait for `schema` by evyatar to handle default, and for `secret` prop + number control etc.

File: modules/atomic-widgets/widgets/atomic-google-maps.php (hunk 6 above)
Comment (blackcolt) [reply modules/atomic-widgets/widgets/atomic-google-maps.php]: Tell me if clearrer now.

File: modules/atomic-widgets/widgets/atomic-google-maps.php (hunk 9 above)
Comment (blackcolt) [reply modules/atomic-widgets/widgets/atomic-google-maps.php]: its save the parmas (a bit faster)

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+		$api_key = $this->get_settings( 'api_key' ) ?? false;
+		$params = [
+			$this->get_settings( 'address' ) ?? esc_html__( 'London Eye, London, United Kingdom', 'elementor' ),
+			$this->get_settings( 'zoom' ) ?? 10,
+			$api_key,
+		];
+
+		if ( $api_key ) {
+			$url = 'https://www.google.com/maps/embed/v1/place?key=%3$s&q=%1$s&amp;zoom=%2$d';
+		}
```
Comment (Omerisra6) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:44]:
```suggestion
		$public_url = 'https://maps.google.com/maps/?q=%1$s&amp;t=m&amp;z=%2$d&amp;output=embed&amp;iwloc=near';
		
			$api_key_url = 'https://www.google.com/maps/embed/v1/place?key=%3$s&q=%1$s&amp;zoom=%2$d';
		$url = ! $api_key ? $public_url : api_key_url;
```

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+		return 'a-google-maps';
+	}
+	protected function render() {
+		$url = 'https://maps.google.com/maps/?q=%1$s&amp;t=m&amp;z=%2$d&amp;output=embed&amp;iwloc=near';
+		$api_key = $this->get_settings( 'api_key' ) ?? false;
+		$params = [
+			$this->get_settings( 'address' ) ?? esc_html__( 'London Eye, London, United Kingdom', 'elementor' ),
+			$this->get_settings( 'zoom' ) ?? 10,
+			$api_key,
+		];
```
Comment (Omerisra6) [inline modules/atomic-widgets/widgets/atomic-google-maps.php:40]:
```suggestion
               $address = $this->get_settings( 'address' ) ?? esc_html__( 'London Eye, London, United Kingdom', 'elementor' );
               $zoom = 	$this->get_settings( 'zoom' ) ?? 10;
		$params = [
                        $address,
		        $zoom,
			$api_key,
		];
```

Then use this attributes in the template.

File: phpcs.xml [hunk 12]
```diff
@@ -5,7 +5,6 @@
 	<arg name="parallel" value="8" />
 
 	<config name="text_domain" value="elementor" />
-
```
Comment (Omerisra6) [inline phpcs.xml:8]: remove

**Review by Omerisra6 — CHANGES_REQUESTED** [review]

File: modules/atomic-widgets/widgets/atomic-google-maps.php [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+			)->set_label( __( 'API Key', 'elementor' ) )
+				->set_placeholder( esc_html__( 'Enter your Google Maps API key', 'elementor' ) ),
+			Textarea_Control::bind_to(
+				'address'
+			)->set_label( __( 'Address', 'elementor' ) )
+			->set_placeholder( esc_html__( 'London Eye, London, United Kingdom', 'elementor' ) ),
+			Textarea_Control::bind_to(
+				'zoom'
+			)->set_label( __( 'Zoom', 'elementor' ) )
+			->set_placeholder( 10 ),
```
Comment (Omerisra6) [inline modules/atomic-widgets/widgets/atomic-google-maps.php]: change to string

**Review by Omerisra6 — CHANGES_REQUESTED** [review]

**Review by Omerisra6 — APPROVED** [review]

**Pull request closed without merging**